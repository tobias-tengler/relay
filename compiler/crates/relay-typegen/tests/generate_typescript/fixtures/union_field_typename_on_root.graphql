fragment a on Query {
  maybeNode {
    __typename
    ... on NonNode {
      name
    }
    ... on Story {
      actorCount
    }
  }
}