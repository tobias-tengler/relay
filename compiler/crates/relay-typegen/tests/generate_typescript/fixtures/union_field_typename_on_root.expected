==================================== INPUT ====================================
fragment a on Query {
  maybeNode {
    __typename
    ... on NonNode {
      name
    }
    ... on Story {
      actorCount
    }
  }
}
==================================== OUTPUT ===================================
import { FragmentRefs } from "relay-runtime";
export type a$data = {
  readonly maybeNode: {
    readonly __typename: "NonNode";
    readonly name: string | null | undefined;
  } | {
    readonly __typename: "Story";
    readonly actorCount: number | null | undefined;
  } | {
    // This will never be '%other', but we need some
    // value in case none of the concrete values match.
    readonly __typename: "%other";
  } | null | undefined;
  readonly " $fragmentType": "a";
};
export type a$key = {
  readonly " $data"?: a$data;
  readonly " $fragmentSpreads": FragmentRefs<"a">;
};
